src/fibsem_tools/io/h5.py

```diff
@@ -0,0 +1,57 @@
+import h5py
+from pathlib import Path
+from typing import Union, Any, Dict
+import warnings
+
+Pathlike = Union[str, Path]
+
+
+def partition_h5_kwargs(**kwargs) -> Dict[str, Any]:
+    """
+    partition kwargs into file-creation kwargs and dataset-creation kwargs
+    """
+    file_kwargs = kwargs.copy()
+    dataset_kwargs = {}
+    for key in ("shape", "dtype", "data", "chunks", "compressor"):
```

Comment:
There is no `compressor` keyword. There are `compression` and `compression_opts` keywords.

* name – Name of dataset to create. May be an absolute or relative path. Provide None to create an anonymous dataset, to be linked into the file later.
* shape – Shape of new dataset (Tuple).
* dtype – Data type for new dataset
* data – Initialize dataset to this (NumPy array).
* chunks – Chunk shape, or True to enable auto-chunking.
* maxshape – Dataset will be resizable up to this shape (Tuple). Automatically enables chunking. Use None for the axes you want to be unlimited.
* compression – Compression strategy. See Filter pipeline.
* compression_opts – Parameters for compression filter.
* scaleoffset – See Scale-Offset filter.
* shuffle – Enable shuffle filter (T/F). See Shuffle filter.
* fletcher32 – Enable Fletcher32 checksum (T/F). See Fletcher32 filter.
* fillvalue – This value will be used when reading uninitialized parts of the dataset.
* track_times – Enable dataset creation timestamps (T/F).
* track_order – Track attribute creation order if True. Default is h5.get_config().track_order.
* external – Store the dataset in one or more external, non-HDF5 files. This should be an iterable (such as a list) of tuples of (name, offset, size) to store data from offset to offset + size in the named file. Each name must be a str, bytes, or os.PathLike; each offset and size, an integer. The last file in the sequence may have size h5py.h5f.UNLIMITED to let it grow as needed. If only a name is given instead of an iterable of tuples, it is equivalent to [(name, 0, h5py.h5f.UNLIMITED)].
* allow_unknown_filter – Do not check that the requested filter is available for use (T/F). This should only be set if you will write any data with write_direct_chunk, compressing the data before passing it to h5py.

Reply:
See https://docs.h5py.org/en/stable/high/group.html#h5py.Group.create_dataset